vertx-web/src/test/java/io/vertx/ext/web/templ/TemplateTest.java

```diff
@@ -72,6 +72,31 @@ private void testRelativeToRoutePath(String pathPrefix) throws Exception {
   }
 
   @Test
+  public void testTemplateHandlerSingleRoute() throws Exception {
+    TemplateEngine engine = new TestEngine(false);
+    router.route("/test-template")
+             .handler(context -> {
+               context.put("foo", "badger");
+               context.put("bar", "fox");
+               context.next();
+             })
+            .handler(TemplateHandler.create(engine, "somedir", "text/html"));
```

Comment:
You're setting a handler twice, which means you will invalidate what happens first (if you run this you should get a warning)